.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+repos:
+  - repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: v19.1.7
+    hooks:
+    - id: clang-format
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+    - id: check-toml
+    - id: forbid-new-submodules
+    - id: end-of-file-fixer
+    - id: check-yaml
```

Comment:
I'm curious - will these `check-` hooks actually do anything? For instance, if I make a new submodule and then submit a PR will it fail CI or?

Reply:
Correct, the check will work as you describe. You can see what it's trying yo do in CI (e.g. https://github.com/ORNL/GridKit/actions/runs/13594004211/job/38006645346#step:4:154). These are mostly legacy checks. We can remove anything we don't need.